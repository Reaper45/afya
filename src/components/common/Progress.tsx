import React from 'react'

import styled from "lib/emotion";

const ProgressWrapper = styled("span")<{ inline?: boolean }>`
  ${props =>
    props.inline &&
    `
  display: flex;
  align-items: center;
  `}
  [data-name="content"] {
    margin-right: 0.6rem;
    font-weight: bold;
  }
`;

const ProgressBar = styled("span")<{ progress?: number; inverted?: boolean; gradient?: boolean; }>`
  ${props =>
    props.inverted &&
    `
    background: ${props.theme.colors.grey};
  `}
  height: 4px;
  width: 100%;
  display: block;
  :after {
    content: "";
    background: ${props => props.gradient ? `linear-gradient(90deg, ${props.theme.colors.primary} 0%, #B794F4 100%)` : props.theme.colors.primary};
    height: 4px;
    width: ${props => props.progress}%;
    border-radius: 2px;
    display: block;
  }
`;

const ProgressDescription = styled("small")`
  color: ${props => props.theme.colors.accent};
  opacity: .66;
  margin-top: .5rem;
  font-size: .7em;
  display: block;
`;

/**
 * Display the progress bar
 * 
 * @param {number} props.progress progress percentage
 */
const Progress: React.FC<{
  description?: string;
  content?: string;
  progress?: number; // progress percentage
  inverted?: boolean;
  gradient?: boolean;
  inline?: boolean;
}> = ({ description, content, progress, inverted, gradient, inline }) => {
  return (
    <ProgressWrapper inline={inline}>
      {content && <span data-name="content">{content}</span>}
      <ProgressBar
        data-name="progress"
        progress={progress}
        inverted={inverted}
        gradient={gradient}
      ></ProgressBar>
      {description && (
        <ProgressDescription data-name="description">
          {description}
        </ProgressDescription>
      )}
    </ProgressWrapper>
  );
};

Progress.defaultProps ={
  inverted: false,
  gradient: false,
  progress: 0,
}

export default Progress;
