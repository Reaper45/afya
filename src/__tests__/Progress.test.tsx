import * as React from "react";
import { shallow } from "enzyme";

import Progress from "components/common/Progress";

it("render correctly", () => {
  const component = shallow(<Progress />);
  expect(component).toMatchSnapshot();
})

it("should render description", () => {
  const component = shallow(<Progress description="24k / 40k patients seen"/>);
  expect(component.find('[data-name="description"]').text()).toEqual(
    "24k / 40k patients seen"
  );
});

it("should render content value", () => {
  const component = shallow(<Progress progress={50} content="50%" />);
  expect(component.find('[data-name="content"]').text()).toEqual("50%");
});
