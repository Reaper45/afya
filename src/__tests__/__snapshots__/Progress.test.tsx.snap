// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<Styled(span)>
  <Styled(span)
    data-name="progress"
    gradient={false}
    inverted={false}
    progress={0}
  />
</Styled(span)>
`;
